botocore/credentials.py

```diff
@@ -55,7 +57,9 @@
 
 logger = logging.getLogger(__name__)
 ReadOnlyCredentials = namedtuple(
-    'ReadOnlyCredentials', ['access_key', 'secret_key', 'token']
+    'ReadOnlyCredentials',
+    ['access_key', 'secret_key', 'token', 'account_id'],
+    defaults=(None,),
```

Comment:
The syntax is a little weird, but `defaults` are applied to the rightmost argument first, so this will only let `account_id` have a default of `None`.